src/utils/components/SidebarEditor/SidebarEditor.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { dump } from 'js-yaml';
+
+import { K8sResourceCommon, YAMLEditor } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  ButtonVariant,
+  Flex,
+  FlexItem,
+  Sidebar,
+  SidebarContent,
+  SidebarPanel,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+
+import Loading from '../Loading/Loading';
+
+import { SidebarEditorContext } from './SidebarEditorContext';
+import { safeLoad } from './utils';
+
+import './sidebar-editor.scss';
+
+type SidebarEditorProps<Resource> = {
+  resource: Resource;
+  onResourceUpdate?: (newResource: Resource) => Promise<Resource | void>;
+  children: (resource: Resource) => React.ReactNode;
+};
+
+const SidebarEditor = <Resource extends K8sResourceCommon>({
+  children,
+  resource,
+  onResourceUpdate,
+}: SidebarEditorProps<Resource>): JSX.Element => {
+  const resourceYAML = React.useMemo(() => dump(resource, { skipInvalid: true }), [resource]);
+  const { showEditor } = useContext(SidebarEditorContext);
+  const [editableYAML, setEditableYAML] = useState('');
+  const editedResource = safeLoad<Resource>(editableYAML);
+
+  const reloadResource = useCallback(() => {
+    setEditableYAML(resourceYAML);
+  }, [resourceYAML]);
+
+  useEffect(() => {
+    reloadResource();
+  }, [reloadResource]);
+
+  const changeResource = (newValue: string) => {
+    setEditableYAML(newValue);
+
+    return {};
+  };
+
+  const edited = editableYAML !== resourceYAML;
```

Comment:
will calculate on each render.. also if used only once you can simply use it at line 76

Reply:
okay. Removed. the save button will be always enabled. Not too bad